doc: Replace nrfjprog to nrfutil in nRF91 docs (#20129)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf91/nrf91_features.rst

Full update
  You can use either a wired or a wireless connection to do a full update of the modem firmware:

>   * When using a wired connection, you can use either the `Programmer app`_, which is part of `nRF Connect for Desktop`_, or `nRF Util`_.

Reviewer: ```suggestion
  * When using a wired connection, you can use either the `Programmer app`_, which is part of `nRF Connect for Desktop`_, or the `nRF Util`_ tool.
```

>   * When using a wired connection, you can use either the `Programmer app`_, which is part of `nRF Connect for Desktop`_, or the `nRF Util`_ tool.

Reviewer 2: ```suggestion
  * When using a wired connection, you can use either the `Programmer app`_, which is part of `nRF Connect for Desktop`_, or `nRF Util's device command <Upgrading modem firmware using J-Link_>`_.
```
For this new link, please add the following in links.txt in line 861:
```
.. _`Upgrading modem firmware using J-Link`: https://docs.nordicsemi.com/bundle/nrfutil/page/nrfutil-device/guides/programming_modem_9160.html
```

>     The  `Programmer app`_ method uses the Simple Management Protocol (SMP) to provide an interface over UART, which enables the device to perform the update.

Reviewer 3: This part is a bit confusing.
To my understanding, both methods use SWD to update the firmware. They both use the same libraries under the hood.

I mean, it is possible to do as described here, but then you are updating via MCUBoot bootloader, which is less common.

For generic nrf91dk docs, I would either just remove this section, or replace with something like "Both methods use SWD to update the firmware".

Author: Discussed offline and replaced with "Both methods use the :term:`Serial Wire Debug (SWD)` interface to update the firmware."

    You can use the Programmer app to perform the update, regardless of the images that are part of the existing firmware of the device.

>     For example, you can update the modem on an nRF9160 DK using the instructions described in the :ref:`nrf9160_updating_fw_modem` section.

Reviewer: ```suggestion
    For example, you can update the modem on an nRF9160 DK using the instructions provided in the :ref:`nrf9160_updating_fw_modem` section.
```
